File: ComparisonExample.cs
using System;

namespace ComparisonExample
{
    public class Bird {
        //Fields
        public int Age = 1;
        public bool Alive = true;

        public string Family = "Cardinal";
        //Methods
        public void Eat() {
            Console.WriteLine("The bird eats");
        }

        public void Breathe() {
            Console.WriteLine("The bird breathes");
        }

        public void Sleep() {
            Console.WriteLine("The bird sleeps");
        }

        public void Die() {
            this.Alive = false;
        }
    }
    class Program
    {
        public static string FirstName = "Aaron";
        public static string LastName = "Smith";
        public static string Name = "Aaron";
        static void Main(string[] args)
        {
            //Initialize objects for comparison
            Bird a = new Bird();
            Bird b = new Bird();
            Bird c = a;

            //Compare references
            if(System.Object.ReferenceEquals(a,b)) {
                Console.WriteLine("Bird a and Bird b have the same reference");
            }
            else {
                Console.WriteLine("Bird a and Bird b have different references");
            }

            //Compare references
            if(System.Object.ReferenceEquals(a,c)) {
                Console.WriteLine("Bird a and Bird c have the same reference");
            }
            else {
                Console.WriteLine("Bird a and Bird c have different references");
            }

            //Compare int
            if(FirstName == Name) {
                Console.WriteLine("The values are equal");
            } else {
                Console.WriteLine("The values are not equal");
            }

            //Compare strings
            if(FirstName == LastName) {
                Console.WriteLine("The values are equal");
            }
            else {
                Console.WriteLine("The values are not equal");
            }
        }
    }
}

File: ComparisonExample.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package comparisonexample;

/**
 *
 * @author Aaron
 */
public class ComparisonExample {
    
    public static String firstName = "Aaron";
    public static String lastName = "Smith";
    public static String name = "Aaron";
    
    public static void main(String[] args) {
        //This example is to show how to make comparisons in Java
        Bird a = new Bird();
        Bird b = new Bird();
        Bird c = a;
        
        if(a == b){
            System.out.println("Bird a and Bird b references are equal");
        }
        else {
            System.out.println("Bird a and Bird b references are not equal");
        }
        
        if(a == c){
            System.out.println("Bird a and Bird c references are equal");
        }
        else {
            System.out.println("Bird a and Bird c references are not equal");
        }
        
        if(firstName.equals(name)) {
            System.out.println("The values are equal");
        }
        else if(!firstName.equals(name)) {
            System.out.println("The values are not equal");
        }
        
        if(firstName.equals(lastName)) {
            System.out.println("The values are equal");
        }
        else if(!firstName.equals(lastName)) {
            System.out.println("The values are not equal");
        }
    }
}
